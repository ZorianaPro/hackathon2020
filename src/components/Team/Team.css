.Team.Team--short {
    border-top: 1px solid var(--liberty-color);
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    padding-top: 1em;
}

.Team.Team--short .Team-Member {
    width: 2em;
    height: 2em;
    border-radius: 50%;
    background-color: var(--white-color);
    display: inline-block;
    margin-right: -0.5em;
    border: 1px solid;
    text-align: center;
    cursor: pointer;
}

.Team.Team--short .Team-Member p {
    margin: 0.4em 0 0 0;
    line-height: 2em;
    vertical-align: center;
    font-size: 0.8em;
}

.Team.Team--short .Team-Join {
    margin-left: 2em;
    cursor: pointer;
    text-transform: uppercase;
    text-decoration: none;
    transition: all ease-in 100ms;
}

.Team.Team--short .Team-Join:hover {
    text-decoration: underline;
}

.Team.Team--full .Team-Members {
    display: flex;
    justify-content: space-between;
    flex-wrap: wrap;
}

.Team.Team--full .Team-Member {
    display: flex;
    min-width: 10em;
    margin: 2em 2em 0 0;
}
.Team.Team--full .Team-Member p {
    border-radius: 50%;
    background-color: var(--white-color);
    border: 1px solid;
    text-align: center;
    width: 3em;
    height: 3em;
    margin: 0 1em 0 0;
    line-height: 3em;
}

.Team.Team--full .Team-Member .Team-Member-Info-Name {
    font-weight: 700;
    margin-bottom: 0.125em;
}

.Team.Team--full .Team-Member .Team-Member-Info-Email {
    font-size: 0.8em;
    text-decoration: none;
}

.Team.Team--full .Team-Member .Team-Member-Info-Position {
    margin-top: 1em;
}