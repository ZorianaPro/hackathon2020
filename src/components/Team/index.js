import React from 'react';
import './Team.css';

const Team = ({
	full = true,
	team
}) => {

	const initials = (firstName, lastName) => {
		return `${firstName.charAt(0).toUpperCase()}${lastName.charAt(0).toUpperCase()}`
	};

	return (
		<div className={`Team ${full ? 'Team--full' : 'Team--short'}`}>
			<div className="Team-Members">
				{ team
				&& (team.members || []).map((member) => {
					return (
						<div className='Team-Member' key={member.id}>
							<p>
								{initials(member.firstName, member.lastName)}
							</p>
							{
								full
								&&
									<div className="Team-Member-Info">
										<div className="Team-Member-Info-Name">
											{`${member.firstName} ${member.lastName}`}
										</div>
										<a href={`mailto:${member.email}`} className="Team-Member-Info-Email">
											{member.email}
										</a>
										<div className="Team-Member-Info-Position">
											{member.position}
										</div>
									</div>
							}
						</div>

					)
				})
				}
			</div>
		</div>
	)
};

export default Team;