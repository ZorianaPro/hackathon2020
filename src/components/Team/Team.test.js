import React from 'react';
import { shallow, mount } from 'enzyme';
import Team from './';

describe('Team', () => {
  it('renders without crashing', () => {
    expect(shallow.bind(shallow, <Team/>)).not.toThrow();
  });

  it('renders as expected', () => {
    expect(shallow(<Team/>)).toMatchSnapshot();
  });

  describe('no members yet', () => {
    let team;

    beforeEach(() => {
      team = mount(<Team/>);
    });
    it('renders default message', () => {
      expect(team.find('.Team--empty').text()).toBe('No members yet');
    });
  });

  describe('has members', () => {
    let team;
    const props = {
      team: [
        {
          id: 'any',
          firstName: 'Zoriana',
          lastName: 'Lesyl',
          email: 'some@random',
          position: 'Manager'
        }, {
          id: 'any',
          firstName: 'Zoriana',
          lastName: 'Lesyl',
          email: 'some@random',
          position: 'Manager'
        }
      ]
    };

    beforeEach(() => {
      team = mount(<Team { ...props }/>);
    });
    it('renders all team members', () => {
      expect(team.find('.Team-Members').children().length).toBe(2);
    });
  });
});