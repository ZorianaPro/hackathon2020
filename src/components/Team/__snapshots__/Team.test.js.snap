// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Team renders as expected 1`] = `
<div
  className="Team Team--full"
>
  <div
    className="Team-Members"
  >
    <div
      className="Team--empty"
    >
      No members yet
    </div>
  </div>
</div>
`;
